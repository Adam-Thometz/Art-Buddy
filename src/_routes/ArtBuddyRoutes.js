import React from "react";
import { Routes, Route } from "react-router-dom";

import Window from "../_components/window/Window";
import LandingPage from "../menu-landing/LandingPage";
import Menu from "../menu-landing/menu/Menu";
import WordToMusic from '../music-decoder/WordToMusic';
import SequenceMaker from '../sequence-maker/SequenceMaker';
import InstrumentId from "../instrument-id/InstrumentId";
  import Learn from "../instrument-id/learn/learn-main/Learn";
  import LearnFamilyPage from "../instrument-id/learn/family/LearnFamilyPage";
  import Instrument from "../instrument-id/learn/instruments/Instrument";
  import Play from '../instrument-id/play/play-main/Play';
  import ListeningSkills from "../instrument-id/play/listening-skills/ListeningSkills";
import ScoreKeeper from '../score-keeper/ScoreKeeper';
import NotFound from "./NotFound";

import urls, { instrumentIdUrls } from "./routeUrls";
import ListeningSkillsTest from "../instrument-id/play/listening-skills/level/ListeningSkillsTest";

const ArtBuddyRoutes = () => {
  return (
    <Routes>
      <Route index element={<LandingPage />} />
      {/* MENUS */}
      <Route path={urls.artGames} element={<Window small page='ART GAMES'><Menu type={'art'} /></Window>} />
      <Route path={urls.musicGames} element={<Window small page='MUSIC GAMES'><Menu type={'music'} /></Window>} />
      <Route path={urls.allGames} element={<Window small page='ALL GAMES'><Menu type={'all'} /></Window>} />

      {/* MUSIC GAMES */}
      {/* Word To Music Decoder */}
      <Route path={urls.wordToMusicUrl} element={<Window page='WORD-TO-MUSIC DECODER'><WordToMusic /></Window>} />
      {/* Sequence Maker */}
      <Route path={urls.sequencerUrl} element={<Window page='SEQUENCE MAKER'><SequenceMaker /></Window>} />
      {/* What's That Instrument? */}
      <Route path={urls.instrumentIdUrl} element={<Window page='INSTRUMENT ID'><InstrumentId /></Window>} />
        {/* Learn */}
        <Route path={instrumentIdUrls.learnUrl} element={<Window page='INSTRUMENT ID: LEARN'><Learn /></Window>}/>
        <Route path={instrumentIdUrls.learnFamilyUrl} element={<Window page='INSTRUMENT ID: LEARN'><LearnFamilyPage /></Window>}/>
        <Route path={instrumentIdUrls.learnInstrumentUrl} element={<Window page='INSTRUMENT ID: LEARN'><Instrument /></Window>}/>
        {/* Play */}
        <Route path={instrumentIdUrls.playUrl} element={<Window page='INSTRUMENT ID: PLAY'><Play /></Window>} />
        <Route path={instrumentIdUrls.playListeningUrl} element={<Window page='INSTRUMENT ID: PLAY'><ListeningSkills /></Window>} />
        <Route path={instrumentIdUrls.playListeningLevelUrl} element={<Window page='Listening Skills Test'><ListeningSkillsTest /></Window>} />
      
      {/* TOOLS */}
      {/* Score Keeper */}
      <Route path={urls.scoreKeeperUrl} element={<Window page='SCORE KEEPER'><ScoreKeeper /></Window>} />
      
      {/* GENERAL 404 HANDLER */}
      <Route path="*" element={<Window small page="UH OH!"><NotFound /></Window>}/>
    </Routes>
  );
};

export default ArtBuddyRoutes;