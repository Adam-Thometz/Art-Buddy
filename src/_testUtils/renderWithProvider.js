import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;