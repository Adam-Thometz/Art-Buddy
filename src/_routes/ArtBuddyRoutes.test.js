import React from "react";

import renderWithProvider from "../_testUtils/renderWithProvider";
import { screen } from "@testing-library/react";

import ArtBuddyRoutes from "./ArtBuddyRoutes";
import urls from './routeUrls';

describe('Art Buddy Routes', () => {
  it('renders without crashing', () => {
    renderWithProvider(<ArtBuddyRoutes />);
  });

  it('matches snapshot', () => {
    const { asFragment } = renderWithProvider(<ArtBuddyRoutes />);
    expect(asFragment()).toMatchSnapshot();
  });
  
  it('renders the music games page', () => {
    renderWithProvider(<ArtBuddyRoutes />, { initialRoutes: [urls.musicGames] });
    expect(screen.getByText('MUSIC GAMES')).toBeInTheDocument();
  });
  
  it('renders the art games page', () => {
    renderWithProvider(<ArtBuddyRoutes />, { initialRoutes: [urls.artGames] });
    expect(screen.getByText('ART GAMES')).toBeInTheDocument();
  });
  
  it('renders all games', () => {
    renderWithProvider(<ArtBuddyRoutes />, { initialRoutes: [urls.allGames] });
    expect(screen.getByText('ALL GAMES')).toBeInTheDocument();
  });
  
  it('renders the word-to-music decoder', () => {
    renderWithProvider(<ArtBuddyRoutes />, { initialRoutes: [urls.wordToMusic] });
    expect(screen.getByText('WORD-TO-MUSIC DECODER')).toBeInTheDocument();
  });
  
  it('renders the sequence maker', () => {
    renderWithProvider(<ArtBuddyRoutes />, { initialRoutes: [urls.sequenceMaker] });
    expect(screen.getByText('SEQUENCE MAKER')).toBeInTheDocument();
  });
  
  it('renders jump into rhythm', () => {
    renderWithProvider(<ArtBuddyRoutes />, { initialRoutes: [urls.jumpIntoRhythm] });
    expect(screen.getByText('JUMP INTO RHYTHM')).toBeInTheDocument();
  });
  
  it('renders the instrument id game', () => {
    renderWithProvider(<ArtBuddyRoutes />, { initialRoutes: [urls.instrumentId] });
    expect(screen.getByText("INSTRUMENT ID")).toBeInTheDocument();
  });
  
  it('renders the score keeper', () => {
    renderWithProvider(<ArtBuddyRoutes />, { initialRoutes: [urls.scoreKeeper] });
    expect(screen.getByText('SCORE KEEPER')).toBeInTheDocument();
  });
  
  it('renders the 404 page', () => {
    renderWithProvider(<ArtBuddyRoutes />, { initialRoutes: ['/gregwerg'] });
    expect(screen.getByText('UH OH!')).toBeInTheDocument();
  });
})