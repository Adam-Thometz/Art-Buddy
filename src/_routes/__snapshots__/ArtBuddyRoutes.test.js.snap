// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Art Buddy Routes matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="Window"
    style="width: 75%;"
  >
    <main
      class="Window"
      style="width: 75%;"
    >
      <section
        class="Welcome"
      >
        <header
          class="WindowNavbar-games"
        >
          <h1
            class="WindowNavbar-label"
          >
            WELCOME TO ART BUDDY!
          </h1>
        </header>
        <article
          class="Welcome-description"
        >
          <p>
            ARTBUDDY is a collection of fun and interactive fine arts games designed specifically with special education goals in mind
          </p>
          <p>
            Click below to get started
          </p>
        </article>
        <button
          class="Button regular"
          style="border-color: #89eb0a;"
        >
          LET'S GO!
        </button>
      </section>
    </main>
  </main>
</DocumentFragment>
`;
